ani-cli

```diff
@@ -177,26 +185,52 @@ decrypt_link() {
 }
 
 # chooses the link for the set quality
-get_video_quality() {
+get_video_link() {
 	dpage_url="$1"
-	video_links=$(decrypt_link "$dpage_url")
+	video_links=$(decrypt_link "$dpage_url")	
+	if printf '%s' "$video_links" | grep -q "mp4"; then 
+		video_url=$(get_video_quality_mp4 "$video_links")
+		idx=1
+	else
+		video_url=$video_links
+		get_video_quality_m3u8
+	fi
+}	
+
+get_video_quality_mp4() {
 	case $quality in
 		best)
-			video_link=$(printf '%s' "$video_links" | head -n 4 | tail -n 1)
+			video_url=$(printf '%s' "$1" | head -n 4 | tail -n 1)
 			;;
 		worst)
-			video_link=$(printf '%s' "$video_links" | head -n 1)
+			video_url=$(printf '%s' "$1" | head -n 1)
 			;;
 		*)
-			video_link=$(printf '%s' "$video_links" | grep -i "${quality}p" | head -n 1)
-			if [ -z "$video_link" ]; then
+			video_url=$(printf '%s' "$1" | grep -i "${quality}p" | head -n 1)
+			if [ -z "$video_url" ]; then
 				err "Current video quality is not available (defaulting to best quality)"
 				quality=best
-				video_link=$(printf '%s' "$video_links" | head -n 4 | tail -n 1)
+				video_url=$(printf '%s' "$1" | head -n 4 | tail -n 1)
 			fi
 			;;
 	esac
-	printf '%s' "$video_link"
+	printf '%s' "$video_url"
+}
+
+get_video_quality_m3u8() {
+	case $quality in
+		worst|360)
+			idx=2;;
+		480)
+			idx=3;;
+		720)
+			idx=4;;
+		*)
```

Comment:
Could this be `best|1080|*`? I know it's redundant, but it would make it way cleaner...

Reply:
Yup.. there should be a default case as per the shellcheck recommendation.. so I put the best|1080p on default

Reply:
@port19x